src/monit/patch/0002-change_monit_alert_log_error.patch

```diff
@@ -0,0 +1,70 @@
+From 97a5defc6a7fcc6a00f691bb5314ceb8fb7704e9 Mon Sep 17 00:00:00 2001
+From: Abhishek Dosi <abdosi@microsoft.com>
+Date: Mon, 26 Oct 2020 11:40:02 -0700
+Subject: [PATCH] Patch on top of commit Patch is addressing these changes:-
+
+a) Make sure error message is loggged if state is changed to fail first time based on fault tolerance condition
+or we have repeat clause enable for alert
+
+b) Enable repeat keyword for alert action . Using this we can log
+syslog error message for persistent failures in repeat window.
+
+c) Make sure log and error message are different string so that we do not rate
+limit error message with log message
+
+Signed-off-by: Abhishek Dosi <abdosi@microsoft.com>
+
+---
+ src/event.c | 6 +++++-
+ src/p.y     | 8 +++++++-
+ 2 files changed, 12 insertions(+), 2 deletions(-)
+
+diff --git a/src/event.c b/src/event.c
+index ed363ee..969568f 100644
+--- a/src/event.c
++++ b/src/event.c
+@@ -336,8 +336,12 @@ static void _handleEvent(Service_T S, Event_T E) {
+                 if (E->state != State_Init || E->state_map & 0x1) {
+                         if (E->state == State_Succeeded || E->state == State_ChangedNot || E->id == Event_Instance || E->id == Event_Action)
+                                 LogInfo("'%s' %s\n", S->name, E->message);
+-                        else
++                        /* Send Error log if state change to failed for 1st time or if we have repeat clause then do periodically */
++                        else if ((E->state_changed) || (E->state == State_Failed && E->action->failed->repeat && E->count % E->action->failed->repeat == 0))
+                                 LogError("'%s' %s\n", S->name, E->message);
```

Comment:
I feel like we should reserve ERROR level messages for errors in Monit. Maybe we can log these alerts at a different level, like ALERT (Monit defines a `LogAlert()` function). Thoughts?

Reply:
Not sure if we have any alert messages in Sonic as of now. Also all our sonic-mgmt test case Filters are on ERR pattern. That will need update if we move to Alert.

Reply:
@yxieca to comment on the use of ALERT-level messages here rather than ERROR (with respect to sonic-mgmt tests).

Reply:
We will go ahead with this as an ERROR-level message.